src/main/java/org/generationcp/middleware/service/impl/dataset/DatasetServiceImpl.java

```diff
@@ -227,6 +238,17 @@ private MeasurementVariable buildSampleColumn() {
 		return sampleColumn;
 	}
 
+	private MeasurementVariable buildStockIdColumn() {
```

Comment:
Generalize buildSampleColumn and reuse it for this case too

Reply:
Done.